sycl/include/CL/sycl/detail/pi.h

```diff
@@ -0,0 +1,245 @@
+static const uint8_t PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE = 3; // LLVM bitcode
+
+// Device binary descriptor version supported by this library.
+static const uint16_t PI_DEVICE_BINARY_VERSION = 1;
+
+// The kind of offload model the binary employs; must be 4 for SYCL
+static const uint8_t PI_DEVICE_BINARY_OFFLOAD_KIND_SYCL = 4;
+
+/// Target identification strings for
+/// pi_device_binary_struct.DeviceTargetSpec
+///
+#define PI_DEVICE_BINARY_TARGET_UNKNOWN "<unknown>"
+#define PI_DEVICE_BINARY_TARGET_SPIRV32 "spir"
+#define PI_DEVICE_BINARY_TARGET_SPIRV64 "spir64";
+
+/// This struct is a record of the device binary information. If the Kind field
+/// denotes a portable binary type (SPIRV or LLVMIR), the DeviceTargetSpec field
+/// can still be specific and denote e.g. FPGA target.
+/// It must match the __tgt_device_image structure generated by
+/// the clang-offload-wrapper tool when their Version field match.
+struct pi_device_binary_struct {
+  /// version of this structure - for backward compatibility;
+  /// all modifications which change order/type/offsets of existing fields
+  /// should increment the version.
+  uint16_t Version;
+  /// the type of offload model the binary employs; must be 4 for SYCL
+  uint8_t Kind;
+  /// format of the binary data - SPIRV, LLVMIR bitcode,...
+  uint8_t Format;
+  /// null-terminated string representation of the device's target architecture
+  const char *DeviceTargetSpec;
+  /// a null-terminated string; target- and compiler-specific options
+  /// which are suggested to use to "build" program at runtime
+  const char *BuildOptions;
+  /// Pointer to the manifest data start
+  const char *ManifestStart;
+  /// Pointer to the manifest data end
+  const char *ManifestEnd;
+  /// Pointer to the target code start
+  const unsigned char *BinaryStart;
+  /// Pointer to the target code end
+  const unsigned char *BinaryEnd;
+  /// the offload entry table (not used, for compatibility with OpenMP)
+  _pi_offload_entry EntriesBegin;
+  _pi_offload_entry EntriesEnd;
+};
+typedef pi_device_binary_struct * pi_device_binary; 
+
+// Offload binaries descriptor version supported by this library.
+static const uint16_t PI_DEVICE_BINARIES_VERSION = 1;
+
+/// This struct is a record of all the device code that may be offloaded.
+/// It must match the __tgt_bin_desc structure generated by
+/// the clang-offload-wrapper tool when their Version field match.
+struct pi_device_binaries_struct {
+  /// version of this structure - for backward compatibility;
+  /// all modifications which change order/type/offsets of existing fields
+  /// should increment the version.
+  uint16_t Version;
+  /// Number of device binaries in this descriptor
+  uint16_t NumDeviceBinaries;
+  /// Device binaries data
+  pi_device_binary DeviceBinaries;
+  /// the offload entry table (not used, for compatibility with OpenMP)
+  _pi_offload_entry *HostEntriesBegin;
+  _pi_offload_entry *HostEntriesEnd;
```

Comment:
> @keryell: This should go in some OpenMP specific struct/classes.

These are opaque for now, and comment tells for compatibility with OpenMP. But we think it may soon be needed for the PI too.